tensordict/nn/distributions/discrete.py

```diff
@@ -87,3 +88,60 @@ def rsample(
         out = d.rsample(sample_shape)
         out.data.copy_((out == out.max(-1)[0].unsqueeze(-1)).to(out.dtype))
         return out
+
+
+class Ordinal(D.Categorical):
+    """A discrete distribution for learning to sample from finite ordered sets.
+
+    It is defined in contrast with the `Categorical` distribution, which does
+    not impose any notion of proximity or ordering over its support's atoms.
+    The `Ordinal` distribution explicitly encodes those concepts, which is
+    useful for learning discrete sampling from continuous sets. See §5 of
+    [Tang & Agrawal, 2020](https://arxiv.org/pdf/1901.10500.pdf) for details.
+
+    Notes:
+        This class is mostly useful when you want to learn a distribution over
+        a finite set which is obtained by discretising a continuous set.
+    """
+
+    def __init__(self, scores: torch.Tensor):
+        """A discrete distribution for learning to sample from finite ordered sets.
+
+        Args:
+            scores: a tensor of shape [..., N] where N is the size of the set which supports the distributions.
+            Typically, the output of a neural network parametrising the distribution.
+        """
+        logits = _generate_ordinal_logits(scores)
+        super().__init__(logits=logits)
+
+
+class OneHotOrdinal(OneHotCategorical):
+    """The one-hot version of the :class:`~.Ordinal` distribution."""
```

Comment:
I think this link will look for `OneHotOrdinal.Ordinal` but i could be wrong (sphinx is still a unpredictable to me)
```suggestion
    """The one-hot version of the :class:`~tensordict.nn.distributions.Ordinal` distribution.
    
    Args:
        scores (torch.Tensor): a tensor of shape [..., N] where N is the size of the set which supports the distributions.
            Typically, the output of a neural network parametrising the distribution.
    """
```